src/index.js

```diff
@@ -346,19 +352,20 @@ class MongoOplogReader extends EventEmitter {
       return startTs.then(ts => {
         debug(`${replSetName} ts: %s`, ts);
         const opts = { since: ts || 1 }; // start where we left off, otherwise from the beginning
-        const oplog = MongoOplog(db, opts);
-        this.oplogs[connStr] = oplog;
-        oplog.on('op', data => {
-          if (data.fromMigrate) return; // ignore shard balancing ops
-          if (data.op === opCode.NOOP) return; // ignore informational no-operation
-          this.processOp(data, replSetName, memberName);
-          this.emit('shard-op', { data, replSetName, memberName });
-        });
-        oplog.on('end', () => {
+        const stream = createOplogStream(db, opts);
+        this.oplogs[connStr] = stream;
+        stream.pipe(through2.obj((data, enc, done) => {
```

Comment:
I will change this to process multiple oplog docs concurrently per worker.

Reply:
Nice!